src/session-pool.ts

```diff
@@ -669,7 +682,7 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     this._traces.delete(session.id);
     this._pendingPrepare++;
     session.type = types.ReadWrite;
-    this._prepareTransaction(session)
+    this._prepareTransaction(session, optimisticLock)
```

Comment:
This doesn't seem right. After releasing the session back to the pool, if this transaction was RW, this is attempting to prepare the next RW transaction on this session. At this point, we don't know if the next transaction will be optimistic or not.

Reply:
Fixed. Using one common session pool for RO, RW and OCC transactions.